[WIP]Listing knative resources (kapis)

Leboudin opened the pull request:
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This PR added apis to list knative resources. Also named an kapi group: serverless. 

This PR is primarily for further discussions.

**Special notes for reviewers**:

May break go mod dependices. 

**Additional documentation, usage docs, etc.**:

Proposal & discussions: [Serverless Support](https://kubesphere.com.cn/forum/d/563-feature-proposal-serverless-support)

calvinyv commented:
@Leboudin welcome to join Ks(KubeSphere) dev family and really appreciate this high value PR, we see you post a proposal on our forum about integrate Knative with Ks, and that's a very good proposal and many areas are covered in Ks, like monitoring, eventing and DevOps, but we hope you could provide more detail about it, something like user story and design doc, as you could see, Ks has excellent user experience and reduce users' learning curve in many place, a very good example is grayscale release function in Ks which is based on Istio, but actually you could see end user doesn't need to know any tech detail about Istio, he/she just case about business and result.
So similar design principle should be apply to other new features in Ks, including Knative, when it happens in Ks, we hope it could have same user experience as as places. 
Maybe you think this make code work complicated and slow progress, but that's the way to build better product, and don't worry about it, we are working as a team, once the user story and design doc is more rich, our user experience designer and developer could join the discussion, we could also hold periodical e-meeting online.
Happy 2019, and best wishes to you and Ks in 2020.

Leboudin commented:
Hi there calvinyv! 

I've created new design proposal located at `docs/en/concepts-and-designs/serverless-design-proposal.md`. For now I've added:

- Overview and definition. Following your instructions, I simplified Knative concepts to make it easier to understand.
- User stories and functionalities.
- Serverless service management API design.

Your and Kubesphere team's advices are valuable, looking forward to have more discussions with you guys.

zryfish commented:
/close
